### PR TITLE
Document exec.d

### PR DESCRIPTION
**Pull request by AidanDelaney**
Provide documentation and examples of how to use exec.d

### CODE REVIEW

**Comment by AidanDelaney** [comment]
I struggled with this last week, so I finally got around to documenting it :P

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md [hunk 1]
```diff
@@ -5,6 +5,95 @@ weight=99
...
 
-If you are familiar with this content and would like to make a contribution, please feel free to open a PR :)
+## Key Points:
+
+    1. Location and Naming: Scripts are placed in the `<layer>/exec.d/` directory within a lunch layer and must be executable. They can have any name.
+
+    2. Script Behavior:
+        * Environment Variables: Scripts can output environment variables in the form of key=value pairs. These variables are added to the application's runtime environment.
+        * File descriptor 3: The output should be directed to file descriptor 3, passed as the first argument to the executable, for environment variable settings. Any other output is ignored.
+        * Exit Code: The scripts should exit with a status code of `0` to indicate success. A non-zero exit code will indicate an error and prevent the application from launching.
```
Comment (natalieparellano) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md:17]:
```suggestion
    * **Inputs**
        * A third open file descriptor (in addition to stdin and stdout).
    * **Outputs**
        * Valid TOML describing environment variables in the form of key=value pairs. These variables are added to the application's runtime environment. The content should be written to file descriptor 3 (see examples for how to do this).
        * Exit Code: The scripts should exit with a status code of `0` to indicate success. A non-zero exit code will indicate an error and prevent the application from launching.
```

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md [hunk 2]
```diff
@@ -5,6 +5,95 @@ weight=99
...
+
+## Use Cases:
+* Dynamic Configuration: Inject configuration values that are determined at runtime.
+* Service Bindings: Configure environment variables based on bound services.
+
+## Implementation:
```
Comment (natalieparellano) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md:23]:
```suggestion
## Implementation Steps:
```

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md [hunk 3]
```diff
@@ -5,6 +5,95 @@ weight=99
...
+* Write Scripts: Create executable scripts within the `<layer>/exec.d/` directory.
+* Set Permissions: Ensure scripts have the appropriate execute permissions (chmod +x).
+* Environment Variables: Use scripts to print `key="value"` pairs to the file descriptor passed as the first argument to the executable.
```
Comment (natalieparellano) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md:26]:
```suggestion
* Environment Variables: Use scripts to print `key="value"` pairs to the third open file descriptor.
```

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md [hunk 4]
```diff
@@ -5,6 +5,95 @@ weight=99
...
+
+### Example
+
+`exec.d` executables can be written in any language.  We provide examples in bash, Go and Python that inject the `EXAMPLE="test"` into the runtime environment.  It is important that environment variables are written to the open file descriptor passed as the first argument to the `exec.d` binary.
```
Comment (natalieparellano) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md:30]:
```suggestion
`exec.d` executables can be written in any language.  We provide examples in bash, Go and Python that inject `EXAMPLE="test"` into the runtime environment.  It is important that environment variables are written to the third open file descriptor.
```

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md [hunk 5]
```diff
@@ -5,6 +5,95 @@ weight=99
...
+	"os"
+)
+
+func main() {
+    // Convert the file descriptor argument to an integer
+    fd, err := strconv.Atoi(os.Args[1])
+    if err != nil {
+        fmt.Println("Invalid file descriptor:", os.Args[1])
+        return
+    }
```
Comment (natalieparellano) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md:58]: I don't think this is used anywhere? The launcher does not actually provide any arguments when invoking exec.d

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md [hunk 6]
```diff
@@ -5,6 +5,95 @@ weight=99
...
+### Example
+
+`exec.d` executables can be written in any language.  We provide examples in bash, Go and Python that inject the `EXAMPLE="test"` into the runtime environment.  It is important that environment variables are written to the open file descriptor passed as the first argument to the `exec.d` binary.
+
+A `bash` example looks as follows:
+```bash
+#!/bin/bash
+
+# Get the file descriptor from the first argument
+FD=$1
```
Comment (natalieparellano) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md:37]:
```suggestion
FD=&3
```
I think this is how you would do this in bash? I would need to try it out to be sure

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md [hunk 7]
```diff
@@ -5,6 +5,95 @@ weight=99
...
+
+    # Write the environment variable to the given file descriptor
+    os.write(fd, b'EXAMPLE="test"\n')
+
+if __name__ == "__main__":
+    if len(sys.argv) != 2:
+        print("Usage: python3 output_example.py <file_descriptor>", file=sys.stderr)
+        sys.exit(1)
+
+    main(sys.argv[1])
```
Comment (natalieparellano) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md:96]: I think this example needs to be updated also but IDK python so 🤷🏼

**Review by natalieparellano — COMMENTED** [review]
Thanks for doing this @AidanDelaney. This is a longstanding gap in our documentation.

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md (hunk 6 above)
Comment (AidanDelaney) [reply content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md:39]: The spec states that the FD is passed as the first argument https://github.com/buildpacks/spec/blob/main/buildpack.md#execd (or am I reading it wrong)?

Comment (natalieparellano) [reply content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md:39]: Hmm... I'm looking at

> 
Input | Description
-- | --
$0 | Absolute path of the executable
†FD 3 | A third open file descriptor

You can ignore the last two lines of the table (`<handle>` and the env var). They are for Windows only.

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md [hunk 8]
```diff
@@ -14,6 +14,7 @@ The [buildpacks `exec.d` interface](https://github.com/buildpacks/spec/blob/main
     2. Script Behavior:
     * **Inputs**
         * A third open file descriptor (in addition to stdin and stdout).
+        * The third open file descriptor is inherited from the calling process.
```
[the remark this answers is not in the document]
Comment (AidanDelaney) [reply content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md]: Done!

**Review by natalieparellano — APPROVED** [review]

File: content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md [hunk 9]
```diff
@@ -5,6 +5,81 @@ weight=99
...
+## Use Cases:
+* Dynamic Configuration: Inject configuration values that are determined at runtime.
+* Service Bindings: Configure environment variables based on bound services.
+
+## Implementation Steps:
+* Write Scripts: Create executable scripts within the `<layer>/exec.d/` directory.
+* Set Permissions: Ensure scripts have the appropriate execute permissions (chmod +x).
+* Environment Variables: Use scripts to print `key="value"` pairs to the third open file descriptor.
+
+### Example
```
Comment (natalieparellano) [inline content/docs/for-buildpack-authors/how-to/write-buildpacks/use-exec.d.md:30]:
```suggestion
### Examples
```

**Pull request merged**